src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -55,7 +55,7 @@ public class DefaultHtmlGenerator : IHtmlGenerator
             _htmlEncoder = htmlEncoder;
 
             // Underscores are fine characters in id's.
-            IdAttributeDotReplacement = "_";
+            IdAttributeDotReplacement = optionsAccessor.Options.HtmlHelperOptions.IdAttributeDotReplacement;
```

Comment:
`IdAttributeDotReplacement` was indeed always static, so giving it an "upgrade" in that it is now a per-MVC instance setting is great.